ts/IterableInboxMessageCell.tsx

```diff
@@ -1,47 +1,117 @@
-"use strict"
-import React, { Component } from "react"
-import { View, Text, StyleSheet } from "react-native"
+"use strict";
+import React, { Component, useState } from "react";
+import { 
+   View, 
+   Text, 
+   StyleSheet, 
+   TouchableOpacity 
+} from "react-native";
 
 type MessageCellProps = {
+   index: number,
    message: {[key: string]: any},
+   handleMessageSelect: Function,
    last: boolean 
 }
 
-const IterableInboxMessageCell = ({ message, last }: MessageCellProps) => {
+const IterableInboxMessageCell = ({ index, message, handleMessageSelect, last }: MessageCellProps) => {
    const unreadIndicator = "\u2022";
-   const messageTitle = message.inboxMetadata.title
-   const messageBody = message.inboxMetadata.subtitle
-   const messageCreatedAt = message.createdAt
+   const messageTitle = message.inboxMetadata.title;
+   const messageBody = message.inboxMetadata.subtitle;
+   const messageCreatedAt = message.createdAt;
+
+   const [active, setActive] = useState(false);
 
    function displayUnreadMessage() {
-      return ( 
-         <View style={last ? styles.lastMessageCell : styles.messageCell}>
+      return (
+         <TouchableOpacity
+            style={(active) ? styles.pressedMessageCell : styles.messageCell}
+            activeOpacity={1}
+            onPress={() => {
+               setActive(!active);
+               handleMessageSelect(index);
+            }}
+         >
             <View style={styles.unreadIndicatorContainer}>
-               <Text style={styles.unreadIndicator}>{unreadIndicator}</Text>
+                  <View style={styles.unreadIndicator}/>
             </View>
             <View style={styles.unreadMessageContainer}>
```

Comment:
I think an abstraction/refactorization should be able to happen here, between a read message and an unread message, the only difference is the unread indicator, so, code wise, there shouldn't need to be duplication here.

The other thing to think about is also having the unread indicator have an on and off state so you can pass along just 1 message cell for either read/unread state.

Reply:
I agree. I will refactor the code to toggle the unread indicator in one display function.

Reply:
I refactored the code to have a single display message function. But, it does implement the dreaded nested tertiary expression.